pages/implementation/start-using-avo-codegen.mdx

```diff
@@ -5,15 +5,59 @@ title: 'Start using Avo Codegen'
 
 import Link from '../../components/Link';
 
-In this guide we'll cover all the required steps to start using Avo in your app or website, both alongside existing tracking code and SDKs and in a new project.
+## What is Codegen
+
+Codegen produces type safe code for implementing analytics. 
+
+- The "data designer" specifies the event structure in the <Link passHref href="/workspace"><a>Avo Tracking Plan</a></Link>, and which platforms should send the event
+- then the developer who implements analytics can use Codegen to implement the analytics calls per each analytics event
+
+## Why use Codegen
+
+Codegen simplifies implementation and makes sure it is correct and consistent across all platforms.
+
+For example, instead of Android calling `analytics.track("game started")` and iOS calling `analytics.track("gameStarted")` , they both call `Avo.gameStarted()`, and the Avo function takes care of the spelling of the event and property names as they get passed into the analytics SDK.
+
+In other words, event and property naming is abstracted entirely away from the event implementation layer, ensuring that events and properties are named the same across all platforms, teams and code bases.
+
+## When use Codegen
+
+Codegen is our take on increasing the robustness of the analytics implementation and simplifying the implementation process.
+
+Adopting Codegen is not required to get value from Avo - many companies have their own ways to achieve robust analytics implementation and prefer to stick to them after adopting Avo and it’s totally fine.
```

Comment:
```suggestion
Adopting Codegen is not required to get value from Avo - many companies have their own ways to achieve robust analytics implementation and prefer to stick to them rather than adopting a new approach.
```